pkg/controller/worker/machines_test.go

```diff
@@ -818,6 +920,23 @@ func expectGetSecretCallToWork(c *mockclient.MockClient, awsAccessKeyID, awsSecr
 		})
 }
 
+// expectDeleteAllCallToWork is the intialization methods which mocks the
+// delete call to delete all awsMachineClass from awsMachineClassList
+func expectDeleteAllCallToWork(c *mockclient.MockClient, awsMachineClassList *machinev1alpha1.AWSMachineClassList) {
+	c.EXPECT().
+		DeleteAllOf(context.TODO(), gomock.Any(), gomock.Any()).
+		Do(func(_ context.Context, object runtime.Object, namespace client.InNamespace) error {
+			mc, ok := object.(*machinev1alpha1.AWSMachineClass)
+			if !ok || mc == nil {
+				return fmt.Errorf("Request is not for desired kind object, hence deletion itsn't accepted")
+			}
```

Comment:
Can we achieve the same result with something like 

```
	c.EXPECT().
		DeleteAllOf(context.TODO(), gomock.AssignableToTypeOf(&machinev1alpha1.AWSMachineClass{}), gomock.Any()).
```

And then gomock will fail when the `DeleteAllOf` is called with the non-expected args.

Reply:
Done here - https://github.com/gardener/gardener-extension-provider-aws/pull/212/commits/bb4f0948c8d40645c1258270c454229d115f6fb3#diff-12a1fa4ba4f4250731a53989c476c9407e832ccecae17ad3fc1b2eee100ad7c3R847.